smartsim/_core/config/config.py

```diff
@@ -161,10 +161,7 @@ def dragon_dotenv(self) -> Path:
 
     @property
     def dragon_server_path(self) -> t.Optional[str]:
-        return os.getenv(
```

Comment:
The `SMARTSIM_DRAGON_SERVER_PATH_EXP` env var was the temporary solution that we came up with for injecting state into a launcher (specifically the `DragonLauncher`) in the API where we assumed all launchers to be stateless.

Basically, if a user created an experiment with `launcher="dragon"` we would silently export this path and then set up the launcher by checking the env var, and then removing it. Ideally a user should never know about its existence.

BUT! Now that we have a way to create a launcher based on the information from an experiment (e.g. `LauncherLike.create` method) we do not need to do this little backdoor trick. Its also nice because it means are code becomes more safe, more separable, and MUCH easier to test!